phc/services/tools.py

```diff
@@ -0,0 +1,281 @@
+        """
+        version_request = {
+            "version": version,
+            "isDefault": is_default,
+        }
+
+        res = self._api_call(
+            f"/v1/trs/v2/tools/{tool_id}/versions",
+            json=version_request,
+            http_verb="POST",
+        )
+        upload_request = {
+            "fileName": source.split("/").pop(),
+            "toolId": res["id"],
+            "version": res["meta_version"],
+        }
+
+        upload_response = self._api_call(
+            "/v1/trs/files", json=upload_request, http_verb="POST"
+        )
+        file_size = os.path.getsize(source)
+        self._api_call_impl(
+            http_verb="PUT",
+            url=upload_response["uploadUrl"],
+            api_path=None,
+            upload_file=source,
+            headers={
+                "Content-Length": str(file_size),
+                "Authorization": None,
+                "LifeOmic-Account": None,
+                "Content-Type": None,
+            },
+        )
+        return res
+
+    def delete(self, tool_id: str, version: str = None) -> bool:
+        """Deletes a tool
+
+        Parameters
+        ----------
+        tool_id : str
+            The tool ID.
+        version : str, optional
+            The version.
+
+        Returns
+        -------
+        bool
+            True if the delete succeeeds, otherwise False
+        """
+        id = f"{tool_id}:{version}" if version else tool_id
+        return (
+            self._api_call(
+                f"/v1/trs/v2/tools/{id}", http_verb="DELETE"
+            ).status_code
+            == 200
+        )
+
+    def get_list(
+        self,
+        tool_class: str = None,
+        organization: str = None,
+        tool_name: str = None,
+        author: str = None,
+        labels: str = None,
+    ) -> ApiResponse:
+        """Fetch a list of tools from the registry
+
+        Parameters
+        ----------
+        tool_class: str, optional
+            The class of the tool, by default None
+        organization: str, optional
+            The organization that owns the tool, by default None
+        tool_name: str, optional
+            The name of the tool, by default None
+        author: str, optional
+            The creator of the tool, by default None
+        label: str, optional
+            A comma delimted list of labels describing the tool, by default None
+
+        Returns
+        -------
+        phc.ApiResponse
+            The list files response
+        """
+        query_dict = {}
+        if tool_class:
+            query_dict["toolClass"] = tool_class
+        if organization:
+            query_dict["organization"] = organization
+        if tool_name:
+            query_dict["toolname"] = tool_name
+        if author:
+            query_dict["author"] = author
+        if labels:
+            query_dict["label"] = labels
+
+        return self._api_call(
+            f"/v1/trs/v2/tools?{urlencode(query_dict)}",
```

Comment:
Do we have paging enabled on this? We may want to eventually create a `phc.easy` module that gets all of the tools, and it'd be nice to already have it at the service level.

Reply:
I have paging, but it's not tokenized like a lot of our paging, it's page_size, page_count paging.  This is to fit in with the ga4gh standard, which is why I'm just returning all for now.  If you like I can add those in with generous defaults.